common/db_store/admin_db.py

```diff
@@ -0,0 +1,200 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2022 Seagate Technology LLC and/or its Affiliates
+
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License,
+# or (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing, please email
+# opensource@seagate.com or cortx-questions@seagate.com.
+
+from enum import Enum
+import pymongo
+from abc import ABC, abstractmethod
+from common.db_store.error import DBError
+
+
+class DBAdmin(ABC):
+
```

Comment:
Add __init__ to both abstract classes to ensure that mandatory parameters are passed from derived classes.

Reply:
Added '__init__' method in both abstract classes. Also added TODO for singleton design pattern ( Same in LLD as well ).